src/main/java/com/github/reducktion/socrates/generator/DenmarkIdGenerator.java

```diff
@@ -0,0 +1,95 @@
+package com.github.reducktion.socrates.generator;
+
+import com.github.reducktion.socrates.extractor.Citizen;
+import com.github.reducktion.socrates.extractor.Gender;
+
+/**
+ * Generates a new CPR for the provided information
+ *
+ * CPR logic:
+ *  * https://en.wikipedia.org/wiki/Personal_identification_number_(Denmark)
+ *  * https://da.wikipedia.org/wiki/CPR-nummer
+ */
+public class DenmarkIdGenerator implements IdGenerator {
```

Comment:
The class is not required to be public, because we can use the static factory in the class `IdGenerator`.
```suggestion
class DenmarkIdGenerator implements IdGenerator {
```